internal/integrationtest/board/board_test.go

```diff
@@ -651,3 +652,137 @@ func TestBoardDetailsListProgrammersFlag(t *testing.T) {
+	require.NoError(t, err)
+	// Verifies boards are returned
+	requirejson.NotEmpty(t, stdout)
+	// Verifies no board has FQBN set since no platform is installed
+	var data []map[string]interface{}
+	err = json.Unmarshal(stdout, &data)
+	require.NoError(t, err)
+	for i := range data {
+		require.NotContains(t, data[i], "fqbn")
+	}
+	names := make([]string, len(data))
+	for i, v := range data {
+		names[i] = v["name"].(string)
+	}
+	require.Contains(t, names, "Arduino Uno")
+	require.Contains(t, names, "Arduino Yún")
+	require.Contains(t, names, "Arduino Zero")
+	require.Contains(t, names, "Arduino Nano 33 BLE")
+	require.Contains(t, names, "Arduino Portenta H7")
+
+	// Search in non installed boards
+	stdout, _, err = cli.Run("board", "search", "--format", "json", "nano", "33")
+	require.NoError(t, err)
+	// Verifies boards are returned
+	requirejson.NotEmpty(t, stdout)
+	// Verifies no board has FQBN set since no platform is installed
+	err = json.Unmarshal(stdout, &data)
+	require.NoError(t, err)
+	for i := range data {
+		require.NotContains(t, data[i], "fqbn")
+	}
+	names = make([]string, len(data))
+	for i, v := range data {
+		names[i] = v["name"].(string)
+	}
+	require.Contains(t, names, "Arduino Nano 33 BLE")
+	require.Contains(t, names, "Arduino Nano 33 IoT")
+
+	// Install a platform from index
+	_, _, err = cli.Run("core", "install", "arduino:avr@1.8.3")
+	require.NoError(t, err)
+
+	stdout, _, err = cli.Run("board", "search", "--format", "json")
+	require.NoError(t, err)
+	requirejson.NotEmpty(t, stdout)
+	// Verifies some FQBNs are now returned after installing a platform
+	err = json.Unmarshal(stdout, &data)
+	require.NoError(t, err)
+	boards := make(map[string]map[string]interface{})
+	for _, v := range data {
+		if _, ok := v["fqbn"]; ok {
+			boards[v["fqbn"].(string)] = v
+		}
+	}
+	require.Len(t, boards, 26)
+	require.Contains(t, boards, "arduino:avr:uno")
+	require.Equal(t, boards["arduino:avr:uno"]["name"], "Arduino Uno")
+	require.Contains(t, boards, "arduino:avr:yun")
+	require.Equal(t, boards["arduino:avr:yun"]["name"], "Arduino Yún")
+
+	stdout, _, err = cli.Run("board", "search", "--format", "json", "arduino", "yun")
+	require.NoError(t, err)
+	requirejson.NotEmpty(t, stdout)
+	err = json.Unmarshal(stdout, &data)
+	require.NoError(t, err)
+	boards = make(map[string]map[string]interface{})
+	for _, v := range data {
+		if _, ok := v["fqbn"]; ok {
+			boards[v["fqbn"].(string)] = v
+		}
+	}
+	require.Contains(t, boards, "arduino:avr:yun")
+	require.Equal(t, boards["arduino:avr:yun"]["name"], "Arduino Yún")
+
+	// Manually installs a core in sketchbooks hardware folder
+	gitUrl := "https://github.com/arduino/ArduinoCore-samd.git"
+	repoDir := cli.SketchbookDir().Join("hardware", "arduino-beta-development", "samd")
+	_, err = git.PlainClone(repoDir.String(), false, &git.CloneOptions{
+		URL:           gitUrl,
+		ReferenceName: plumbing.NewTagReferenceName("1.8.11"),
+	})
+	require.NoError(t, err)
```

Comment:
We may create a helper function to clone a git repository, so we don't have to repeat this code every time.